voicevox_engine/core/core_wrapper.py

```diff
@@ -384,132 +385,156 @@ def load_core(core_dir: Path, use_gpu: bool) -> CDLL:
-        POINTER(c_long),
-        POINTER(c_float),
-    )
-    core_cdll.yukarin_sa_forward.restype = c_bool
-
-
-def _type_decode_forward(core_cdll: CDLL) -> None:
-    """コアDLL `decode_forward` 関数を型付けする"""
-    core_cdll.decode_forward.argtypes = (
-        c_int,
-        c_int,
-        POINTER(c_float),
-        POINTER(c_float),
-        POINTER(c_long),
-        POINTER(c_float),
-    )
-    core_cdll.decode_forward.restype = c_bool
-
-
-def _type_predict_sing_consonant_length_forward(core_cdll: CDLL) -> None:
-    """コアDLL `predict_sing_consonant_length_forward` 関数を型付けする"""
-    core_cdll.predict_sing_consonant_length_forward.argtypes = (
-        c_int,
-        POINTER(c_long),
-        POINTER(c_long),
-        POINTER(c_long),
-        POINTER(c_long),
-        POINTER(c_long),
-    )
-    core_cdll.predict_sing_consonant_length_forward.restype = c_bool
-
-
-def _type_predict_sing_f0_forward(core_cdll: CDLL) -> None:
-    """コアDLL `predict_sing_f0_forward` 関数を型付けする"""
-    core_cdll.predict_sing_f0_forward.argtypes = (
-        c_int,
-        POINTER(c_long),
-        POINTER(c_long),
-        POINTER(c_long),
-        POINTER(c_float),
-    )
-    core_cdll.predict_sing_f0_forward.restype = c_bool
-
-
-def _type_predict_sing_volume_forward(core_cdll: CDLL) -> None:
-    """コアDLL `predict_sing_volume_forward` 関数を型付けする"""
-    core_cdll.predict_sing_volume_forward.argtypes = (
-        c_int,
-        POINTER(c_long),
-        POINTER(c_long),
-        POINTER(c_float),
-        POINTER(c_long),
-        POINTER(c_float),
-    )
-    core_cdll.predict_sing_volume_forward.restype = c_bool
-
-
-def _type_sf_decode_forward(core_cdll: CDLL) -> None:
-    """コアDLL `sf_decoder_forward` 関数を型付けする"""
-    core_cdll.sf_decode_forward.argtypes = (
-        c_int,
-        POINTER(c_long),
-        POINTER(c_float),
-        POINTER(c_float),
-        POINTER(c_long),
-        POINTER(c_float),
-    )
-    core_cdll.sf_decode_forward.restype = c_bool
-
-
-def _type_last_error_message(core_cdll: CDLL) -> None:
-    """コアDLL `last_error_message` 関数を型付けする"""
-    core_cdll.last_error_message.restype = c_char_p
-
-
-def _type_load_model(core_cdll: CDLL) -> None:
-    """コアDLL `load_model` 関数を型付けする"""
-    core_cdll.load_model.argtypes = (c_long,)
-    core_cdll.load_model.restype = c_bool
-
-
-def _type_is_model_loaded(core_cdll: CDLL) -> None:
-    """コアDLL `is_model_loaded` 関数を型付けする"""
-    core_cdll.is_model_loaded.argtypes = (c_long,)
-    core_cdll.is_model_loaded.restype = c_bool
-
-
-def _type_supported_devices(core_cdll: CDLL) -> None:
-    """コアDLL `supported_devices` 関数を型付けする"""
-    core_cdll.supported_devices.restype = c_char_p
-
-
-def _type_finalize(core_cdll: CDLL) -> None:
-    """コアDLL `finalize` 関数を型付けする"""
-    core_cdll.finalize.restype = None
+_C_TYPE = (
+    type[c_bool]
+    | type[c_int]
+    | type[c_long]
+    | type[c_float]
+    | type[c_char_p]
+    | type["_Pointer[c_long]"]
+    | type["_Pointer[c_float]"]
```

Comment:
記法をよく知らないので間違ってるかもですが、こうでも行けるかも。
```suggestion
    | type[POINTER(c_long)]
    | type[POINTER(c_float)]
```
（こうすると`from ctypes import _Pointer  # noqa: F401`が不要になる）

Reply:
> こうでも行けるかも。

最初は提案の通り書いていたのですが、mypy のチェックが通らなかったのでこの形になりました（`type[]` の中に式は書けないというようなことを言われました。型名やクラス名を書かないといけないようです）。VSCode の拡張が最終的に書いた型のように表示していたので、その通りに書いたらチェックが通りました

Reply:
あれ、なるほどです！！
確認できてなかったかもです、失礼しました！！🙇